core/src/utilities/helpers/storage-helper.js

```diff
@@ -0,0 +1,164 @@
+
+  checkStorageBrowserSupport(){
+    if (!this.browseSupported){
+      throw "Browser does not support local storage"
+    }
+  }
+
+  process(microfrontendId, hostname, id, operation, params){
+    try{
+        this.checkInit();
+        this.checkStorageBrowserSupport();
+        const operationFunction = this[operation];
+        if (typeof operationFunction !== 'function'){
+          throw operation + " is not a supported operation for the storage";
+        }
+        const result = operationFunction.bind(this, this.cleanHostname(hostname), params)();
+        this.sendBackOperation(microfrontendId, id, 'OK', result);
+      }catch(error){
+        console.log(error);
+        this.sendBackOperation(microfrontendId, id, 'ERROR', error);
+      }
+  }
+
+  cleanHostname(hostanme){
+    return hostanme.replace('http://','').replace("https://",'');
+  }
+
+  setItem(hostname, params) {
+    this.checkKey(params);
+    const value = this.stringifyValue(params.value);
+    const key = this.buildKey(hostname, params.key);
+    this.storage.setItem(key, value);
+  }
+
+  getItem(hostname, params){
+    this.checkKey(params);
+    const key = this.buildKey(hostname, params.key);
+    const item = this.storage.getItem(key);
+    if (item){
+      return this.parseJsonIfPossible(item);
+    }else{
+      return undefined;
+    }
+  }
+
+  buildKey(hostname, subKey){
+    return  this.buildPrefix(hostname) + subKey.trim();
+  }
+
+  buildPrefix(hostname){
+    return "Luigi#"+ hostname + "#";
+  }
+
+  removeItem(hostname, params){
+    this.checkKey(params);
+    const key = this.buildKey(hostname, params.key);
+    const item = this.storage.getItem(key);
+    if (item){
+      this.storage.removeItem(key);
+      return item;
+    }else{
+      return undefined;
+    }
+  }
+
+  clear(hostname, params) {
+    const keyPrefix =  this.buildPrefix(hostname);
+    Object.keys(this.storage)
+      .filter(key => key.startsWith(keyPrefix))
+      .forEach(key => this.storage.removeItem(key));
+  }
+
+  has(hostname, params){
+    this.checkKey(params);
+    const key = this.buildKey(hostname, params.key);
+    const item = this.storage.getItem(key);
+    if (item){
+      return true;
+    }else{
+      return false;
+    }
+  }
+
+  getAllKeys(hostname, params){
+    const keyPrefix =  this.buildPrefix(hostname);
+    return Object.keys(this.storage)
+      .filter(key => key.startsWith(keyPrefix))
+      .map(key=> key.substring(keyPrefix.length));
+  }
+
+  checkKey(params){
+    if (!params.key || params.key.trim().length === 0){
+      throw "Missing key, we cannot execute storage operation";
+    }
+  }
+
+  isValueString(value) {
+    return typeof value === 'string' || value instanceof String;
+  }
+
+  parseJsonIfPossible(text){
+    try {
+      return JSON.parse(text);
+    } catch (e) {
+      return text;
+    }
+  }
+
+  stringifyValue(value){
+    if (!value){
+      throw "Value is empty";
+    }
+    if (this.isValueString(value)){
```

Comment:
I felt it is not necessary to be another function, especially it is one line and not used in the other place.

Reply:
isValueString function removed